verifier/verifier.go

```diff
@@ -518,6 +544,40 @@ func verifyAuthenticTimestamp(outcome *notation.VerificationOutcome) *notation.V
 	}
 }
 
+func verifyRevocation(outcome *notation.VerificationOutcome, client *http.Client, logger log.Logger) *notation.ValidationResult {
+	r := revocation.New(client)
+	authenticSigningTime := outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningTime
+	// TODO use authenticSigningTime from signerInfo
+	// https://github.com/notaryproject/notation-core-go/issues/38
+	revocationErrs := r.Validate(outcome.EnvelopeContent.SignerInfo.CertificateChain, authenticSigningTime)
+
+	result := &notation.ValidationResult{
+		Type:   trustpolicy.TypeRevocation,
+		Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+	}
+
+	// Log all non-nil errors as debug
+	for i, serverErrs := range revocationErrs {
+		for _, err := range serverErrs {
+			if err != nil {
+				logger.Debugf("error for certificate #%d in chain: %v", (i + 1), err)
+			}
+		}
+	}
+
+	switch revocation.ResultFromErrors(revocationErrs) {
+	case revocation.OK:
+		logger.Debug("no important errors encountered while checking revocation, status is OK")
```

Comment:
Can we combine revocationErrs and the result (`revocation.*`) into a struct so that we don't need to make multiple calls to `revocation` package? Having a simpler package interface would be ideal.

Reply:
If we did that, the notation-core-go functionality would be coupled to the way that we have specified for errors to be merged. The goal was to abstract the "result" from the individual errors. This was the consensus after a few different discussions and design changes in the notation-core-go PR. What is the downside of making these calls separately?

Reply:
I posted few comments on https://github.com/notaryproject/notation-core-go/pull/134 that are related to simplifying the OCSP status result as well.

> What is the downside of making these calls separately?

That increases the surface area of the package. Larger surface area means more maintenance burden.